### PR TITLE
Reigns v.2

### PR DESCRIPTION
**Pull request by AymericG**
1. Converts to typescript
2. Adds the ability to pass game json file as toolbar parameter

### CODE REVIEW

File: elements/reigns/src/fresco.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+interface IToolbarButton {
```
Comment (maxired) [inline elements/reigns/src/fresco.d.ts:1]: should this file be somewhere else ?

File: elements/reigns/package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "license": "MIT",
   "scripts": {
     "dev": "vite",
-    "build": "vite build",
+    "build": "tsc && vite build",
```
[the remark this answers is not in the document]
Comment (staff0rd) [reply elements/reigns/package.json:8]: `noEmit` is here https://github.com/fres-co/fresco-community/blob/285bd3637aadaf3a9c3e3b1e31e172c0ceafa46a/elements/reigns/tsconfig.json#L16

File: elements/reigns/src/App.tsx [hunk 3]
```diff
@@ -22,17 +21,25 @@ const useFresco = function () {
 
       const defaultState = {
         selectedCard: null,
-        phase: NOT_STARTED,
+        phase: GamePhase.LOADING,
         stats: [],
+        gameUrl: 'https://localhost:3001/games/gdpr.json',
```
Comment (staff0rd) [inline elements/reigns/src/App.tsx:26]: should probably be a relative path, without protocol/host/port?

File: elements/reigns/src/App.tsx [hunk 4]
```diff
@@ -22,17 +21,25 @@ const useFresco = function () {
...
+          ui: { type: "string" },
+          property: "gameUrl",
+        }]});
     });
   }, []);
 
   const store = useStore();
   const updateFrescoState = () => {
-    const state = store.getState();
+    const state = store.getState() as IAppState;
```
Comment (staff0rd) [inline elements/reigns/src/App.tsx:42]: `const store = useStore<IAppState>();`

**Comment by staff0rd** [comment]
> Adds the ability to pass game json file as toolbar parameter

README update?

File: elements/reigns/src/features/game/types.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import { GamePhase } from '../../constants';
+
+export interface IStat {
```
[the remark this answers is not in the document]
Comment (staff0rd) [reply elements/reigns/src/features/game/types.ts:3]: use `type` instead? :)

File: elements/reigns/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (AymericG) [reply elements/reigns/package.json:8]: This is done by the vite generator.

File: elements/reigns/src/fresco.d.ts (hunk 1 above)
Comment (AymericG) [reply elements/reigns/src/fresco.d.ts:1]: I added a JIRA ticket to extract the sdk and its typings into a package.

https://mobitroll.atlassian.net/browse/FRES-1144?atlOrigin=eyJpIjoiOGQ3ZGY2MmRlMGU4NDA4ZWI5ODM0YzVkYTQ2ZTI4MzAiLCJwIjoiaiJ9

**Comment by AymericG** [comment]
> Does it also takes care of `Fix deployments of .png assets of the fresco-community project` ?

I don't know, maybe. We will know once this PR is merged.

**Comment by AymericG** [comment]
I believe I have addressed all comments.

**Review by staff0rd — APPROVED** [review]

**Pull request merged**